### ISSUE TITLE
Add explicit version information to required packages for CMake

### ISSUE THREAD

**Issue by trisyoungs** [trisyoungs/duq#1884]
**Describe the bug**
Having just tried to compile Dissolve on my laptop (OpenSuSE 15.4, Qt6 6.2) I discovered that the new usage of `qt_generate_deploy_qml_app_script` depends on Qt6 >=6.7.

Our current `find_package` calls do not have any version requirements set throughout, so we should add these.

- [ ] Qt >= 6.7
- [ ] ANTLR4 (Java) >= 4
- [ ] ANTLR runtime >= 4.11

**Comment by trisyoungs** [comment]
Building against Qt 6.7 on my machines (OpenSuSE) appears to be broken...

During configuration:
```
CMake Warning at /usr/lib64/cmake/Qt6Qml/Qt6QmlMacros.cmake:3393 (message):
  The qml plugin 'qquicklayoutsplugin' is a dependency of 'dissolve-gui', but
  the link target it defines (Qt6::qquicklayoutsplugin) does not exist in the
  current scope.  The plugin will not be linked.
Call Stack (most recent call first):
  /usr/lib64/cmake/Qt6Core/Qt6CoreMacros.cmake:678 (qt6_import_qml_plugins)
  /usr/lib64/cmake/Qt6Core/Qt6CoreMacros.cmake:678 (cmake_language)
  /usr/lib64/cmake/Qt6Core/Qt6CoreMacros.cmake:774 (_qt_internal_finalize_executable)
  /usr/lib64/cmake/Qt6Core/Qt6CoreMacros.cmake:590:EVAL:1 (qt6_finalize_target)
  CMakeLists.txt:DEFERRED
```
... and many other plugin libraries related to qml.

And during compilation:

```
/home/tris/src/dissolve/build-release/src/gui/models/moc_sortFilterProxy.cpp:131:69: error: static assertion failed: NOTIFY signal read does not exist in class (or is private in its parent)
  131 | static_assert(qxp::is_detected_v<has_nullary_read, SortFilterProxy> || qxp::is_detected_v<has_unary_read, SortFilterProxy>,
```

**Comment by trisyoungs** [comment]
But it seems building against 6.6.3 works just fine without errors. So, errr, maybe ignore all of this.
